### PR TITLE
planner: rename optimizer hints

### PR DESCRIPTION
**Pull request by foreyes**
### What problem does this PR solve? 

Support optimizer hint alias `HASH_JOIN: TIDB_HJ`, `SM_JOIN: TIDB_SMJ`, `INL_JOIN: TIDB_INLJ`.

Rename optimizer hints `TIDB_HASHAGG() -> HASH_AGG`, `TIDB_STREAMAGG() -> STREAM_AGG`, and remove brackets.

### What is changed and how it works?

Add  `HintHJ`, `HintSMJ`, `HintINLJ` have the same effect as `TIDB` hints.

We still keep TIDB style hints, for example, `TIDB_HJ` will have the same HintName as  `HASH_JOIN` now.

Rename aggregation hints in parser and TiDB.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variables name changed.

Related changes

- Related Parser PR: https://github.com/pingcap/parser/pull/424

### CODE REVIEW

**Comment by foreyes** [comment]
Idc fails because use `go mod edit -replace`, this will be fixed after parser PR is merged.

**Comment by foreyes** [comment]
PTAL. @lzmhhh123 @lamxTyler @XuHuaiyu

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -1941,15 +1947,15 @@ func (b *PlanBuilder) pushTableHints(hints []*ast.TableOptimizerHint) {
 	var preferAggType uint
 	for _, hint := range hints {
 		switch hint.HintName.L {
-		case TiDBMergeJoin:
+		case TiDBMergeJoin, HintSMJ:
```
Comment (lzmhhh123) [inline planner/core/logical_plan_builder.go]: It seems `TiDBMergeJoin` is duplicated. Shall we consider removing it? Also take `TiDBIndexNestedLoopJoin` and  `TiDBHashJoin` into consideration.

**Comment by foreyes** [comment]
The solution of `TIDB_HJ -> HASH_JOIN` is not looks well because of the error and warning message. But I think we can improve it by solving this Issue: https://github.com/pingcap/tidb/issues/11632

**Comment by foreyes** [comment]
Ready for review, PTAL. @lzmhhh123

**Comment by foreyes** [comment]
I think so, I will file an issue for it, like https://github.com/pingcap/tidb/issues/11632 @lzmhhh123

**Comment by foreyes** [comment]
PTAL again. Just change some warning messages. @lamxTyler

**Pull request merged**